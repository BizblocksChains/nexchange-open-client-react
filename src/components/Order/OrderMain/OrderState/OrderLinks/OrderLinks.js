import React from 'react';
import config from 'Config';
import getBlockchainUrl from 'Utils/getBlockchainUrl';
import styles from './OrderLinks.scss';
import { I18n } from 'react-i18next';

const OrderLinks = props => (
  <I18n ns="translations">
    {t => (
      <div className={styles.links}>
        {props.order && (
          <a href={`${config.API_BASE_URL}/orders/${props.order.unique_reference}`} className={`btn ${styles.btn}`} target="_blank">
            {t('order.api')}
          </a>
        )}

        {props.coin &&
          props.txId && (
            <a href={getBlockchainUrl(props.coin, props.txId)} className={`btn ${styles.btn}`} target="_blank">
              {t('order.blockchain')}
            </a>
          )}
      </div>
    )}
  </I18n>
);

export default OrderLinks;
