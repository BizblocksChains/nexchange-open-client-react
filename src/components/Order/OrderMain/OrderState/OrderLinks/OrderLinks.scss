.btn.btn.btn {
  color: $green;
  background-color: #fff;
  border: 1px solid $green;
  padding: 6px 10px;
  margin: 10px 0;
  font-size: 12px;
  width: 100%;
  white-space: normal;

  &:hover {
    box-shadow: none;
  }

  &:focus,
  &:hover,
  &:active {
    color: $green;
    background-color: #fff;
    border: 1px solid $green;
  }

  @media (min-width: $screen-sm-min) {
    margin: 10px;
    max-width: 280px;
  }
}
