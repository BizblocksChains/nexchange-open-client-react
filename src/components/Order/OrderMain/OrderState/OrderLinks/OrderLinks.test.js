import React from 'react';
import { shallow } from 'enzyme';
import OrderLinks from './OrderLinks';
import order from 'Mocks/order';

describe('OrderLinks', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(<OrderLinks txId="123" coin="BTC" order={order} />).dive();
    expect(wrapShallow).toMatchSnapshot();
  });
});
