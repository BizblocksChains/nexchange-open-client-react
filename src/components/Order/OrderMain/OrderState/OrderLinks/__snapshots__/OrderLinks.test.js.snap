// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderLinks renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="btn btn"
          href="https://api.nexchange.io/en/api/v1/orders/OV2O6C"
          target="_blank"
>
          order.api
</a>,
        <a
          className="btn btn"
          href="https://blockchain.info/tx/123"
          target="_blank"
>
          order.blockchain
</a>,
      ],
      "className": "links",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "order.api",
          "className": "btn btn",
          "href": "https://api.nexchange.io/en/api/v1/orders/OV2O6C",
          "target": "_blank",
        },
        "ref": null,
        "rendered": "order.api",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "order.blockchain",
          "className": "btn btn",
          "href": "https://blockchain.info/tx/123",
          "target": "_blank",
        },
        "ref": null,
        "rendered": "order.blockchain",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="btn btn"
            href="https://api.nexchange.io/en/api/v1/orders/OV2O6C"
            target="_blank"
>
            order.api
</a>,
          <a
            className="btn btn"
            href="https://blockchain.info/tx/123"
            target="_blank"
>
            order.blockchain
</a>,
        ],
        "className": "links",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "order.api",
            "className": "btn btn",
            "href": "https://api.nexchange.io/en/api/v1/orders/OV2O6C",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "order.api",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "order.blockchain",
            "className": "btn btn",
            "href": "https://blockchain.info/tx/123",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "order.blockchain",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
